### PR TITLE
1.0.0-dev

### PR DESCRIPTION
**Pull request by lordrhodos**
break out the migration functionality as own library from the `starlit/db` package.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "authors": [
+        {
+            "name": "Starweb AB"
+        }
+    ],
+    "require": {
+        "php": ">=7.1",
+        "ext-pdo_mysql": "*",
+        "starlit/db": "^0.12.2"
+    },
```
Comment (felixsand) [inline composer.json:17]: Should we also add the `branch-alias` here?

File: src/Migrator.php [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+ * Class for handling migration between different database versions.
+ *
+ * @author Andreas Nilsson <http://github.com/jandreasn>
+ */
+class Migrator
+{
+    /**
+     * @const string
+     */
+    const DIRECTION_UP = 'up';
```
Comment (felixsand) [inline src/Migrator.php]:
```suggestion
    private const DIRECTION_UP = 'up';
```

File: src/Migrator.php [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+    /**
+     * @const string
+     */
+    const DIRECTION_DOWN = 'down';
```
Comment (felixsand) [inline src/Migrator.php]:
```suggestion
    private const DIRECTION_DOWN = 'down';
```

Comment (felixsand) [reply src/Migrator.php]: Also, no PHPDocs needed for constants

File: src/Migrator.php [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+                $migrations[$migrationNumber] = $migration;
+            }
+        }
+
+        return $migrations;
+    }
+
+    private function shouldMigrationBeMigrated(AbstractMigration $migration, int $toNumber, string $direction): bool
+    {
+        if (($direction === self::DIRECTION_UP
```
Comment (felixsand) [inline src/Migrator.php]: This if is way too complex 😱 
Maybe we could break out this logic into `shouldMigrateUp($migration, $toNumber, $direction)` and `shouldMigrateDown($migration, $toNumber, $direction)` instead?

File: src/Migrator.php [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+                $this->addInfo(sprintf(' - Migrating down %d...', $migration->getNumber()));
+                $migration->down();
+                $this->deleteMigratedMigration($migration);
+
+            }
+        }
+    }
+
+    /**
+     * @param string $info
```
Comment (felixsand) [inline src/Migrator.php]: PHPDocs can be removed here I think

**Review by felixsand — CHANGES_REQUESTED** [review]
Really nice!
Just a couple of adjustments that I think that we might want to add now that we're breaking this out

File: src/Migrator.php (hunk 4 above)
Comment (lordrhodos) [reply src/Migrator.php]: yes why not simplify this while working at it anyway ;-)

**Review by felixsand — DISMISSED** [review]
Perfect! 🎉

**Pull request merged**